File: api/WebApp.Domain/Entities/Project.cs
using NodaTime;

namespace WebApp.Domain.Entities;

public sealed record Project : ISoftDelete
{
    public Instant CreatedTime { get; init; }
    public NamespaceId NamespaceId { get; init; }
    public Namespace Namespace { get; init; } = null!;
    public ProjectId Id { get; init; }
    public string Name { get; init; } = null!;
    public string? Summary { get; init; }
    public string? About { get; init; }
    public string Identifier { get; init; } = null!;
    public Instant? DeletedTime { get; init; }
    public ICollection<ProjectMember> ProjectMembers { get; init; } = null!;
}

File: www/src/lib/models/project.ts
import type { Namespace } from './namespace';
import type { Role } from './role';
import type { Tag } from './tag';
import type { User } from './user';

export interface Project {
	createdTime: string;
	namespaceId: string;
	namespace: Namespace;
	id: string;
	name: string;
	summary: string | null;
	descriptionJson: string | null;
	identifier: string;
	deletedTime: string | null;
	projectMembers: string;
	tags: Tag[];
}

export interface ProjectMember {
	createdTime: string;
	id: string;
	projectId: string;
	project: Project;
	userId: string;
	user: User;
	roleId: string;
	role: Role;
}
